### DISCUSSION TITLE
Understanding the need for `setCoords`

### DISCUSSION THREAD

**Discussion by drinkspiller** [fabricjs/fabric.js#10234] [General]
Regarding the first [Gotcha in the docs "Objects are no more selectable - setCoords"](http://fabricjs.com/fabric-gotchas)  — can someone please help me understand why updating the coordinates isn't handled internally whenever left and/or top are set (like what happens automatically when adjustments are made via the dragging the object or its controls) ? What is a scenario where a developer would *not* want oCoords and aCoords automatically updated? Thanks!

**Comment by asturur** [comment]
When dragging an object or when modifying it with the controls, setCoords is indeed called. ( look at canvas._performTransformAction function )
When selecting an object setCoords is called ( look at canvas._setActiveObject function ).

if you take an object on the canvas and you do:
```
myRect.left = 5
```
Fabricjs does not do anything automatic.

A scenario when you do not want setCoords updated automatically is when you are doing a lot of stuff and you do not want extra code to run that you do not need.

Js is pretty flexible, if you want rewrite the FabricObject.set method in your project and make it trigger a setCoords every time one position property is updated:

- angle
- left
- top
- scaleX
- scaleY
- skewX
- skewY
- originX
- originY
- width
- height
- strokeWidth
(maybe i m forgetting some )
And if you don't find any performance issue leave it in that way.

Which are the situations in which you are out of sync with the coords and you need to call it manually?

**Reply by drinkspiller** [reply]
Thanks for the reply, I appreciate it!

> when you are doing a lot of stuff and you do not want extra code to run that you do not need.

I guess that it what I was asking — what is a scenario where not running the extra code is preferential to an object actually being selectable/hoverable in the place it was positioned?

Indeed I'm familiar with how to modify the code to do this, I'm just trying to understand the context. It seems to me that a developer who has set the top/left position of an object would expect it to be selectable and hoverable in that position without any additional steps. That's certainly what I expected. Thanks again!

**Reply by asturur** [reply]
One situation is animations in which you want to animate many objects.
In general a non selected object has no business to have updated controls coordinates
Or when you do a zoom and pan operation if you have many object on the canvas you do not really want to call setCoords for each of them ( to have updated controls ).

Is the automatism part that is in my opinion is not welcome.
Yes a developer may want everything to work automatically but other would like to have the flexibility to run the code when needed and maintainers to have less things to keep up as possible.
